<sdf version="1.6">
  <world name="multiVessel_world">
    <scene>
      <sky>
        <clouds>
          <speed>12</speed>
        </clouds>
      </sky>
      <grid>0</grid>
      <origin_visual>0</origin_visual>
    </scene>
    
    <spherical_coordinates>
      <surface_model>EARTH_WGS84</surface_model>
      <world_frame_orientation>ENU</world_frame_orientation>
      <latitude_deg>18.254561315779867</latitude_deg>
      <longitude_deg>-64.96912326999195</longitude_deg>
      <elevation>0.0</elevation>
      <heading_deg>0.0</heading_deg>
    </spherical_coordinates>
    
    <include>
      <uri>model://sun</uri>
    </include>
    <gui fullscreen="0">
      <camera name="user_camera">
        <pose>-478.101746 148.200836 13.203143 0.0 0.248344 2.936862</pose>
        
        
        <view_controller>orbit</view_controller>
      </camera>
      
    </gui>
    <model name="ocean_waves">
      <static>true</static>
      <plugin filename="libWavefieldModelPlugin.so" name="wavefield_plugin">
        <static>false</static>
        <update_rate>30</update_rate>
        <size>1000 1000</size>
        <cell_count>50 50</cell_count>
        <wave>
          <model>PMS</model>
          <period>5</period>
          <number>3</number>
          <scale>2.5</scale>
          <gain>0.1</gain>
          <direction>1.0 0.0</direction>
          <angle>0.4</angle>
          <tau>2.0</tau>
          <amplitude>0.0</amplitude>
          
          <steepness>0.0</steepness>
        </wave>
      </plugin>
      <link name="ocean_waves_link">
        <visual name="ocean_waves_visual">
          <plugin filename="libWavefieldVisualPlugin.so" name="ocean_waves_visual_plugin">
            <enableRtt>true</enableRtt>
            <rttNoise>0.1</rttNoise>
            <refractOpacity>0.2</refractOpacity>
            <reflectOpacity>0.2</reflectOpacity>
            <wave>
              <model>PMS</model>
              <period>5</period>
              <number>3</number>
              <scale>2.5</scale>
              <gain>0.1</gain>
              <direction>1.0 0.0</direction>
              <angle>0.4</angle>
              <tau>2.0</tau>
              <amplitude>0.0</amplitude>
              
              <steepness>0.0</steepness>
            </wave>
          </plugin>
          <geometry>
            <mesh>
              <scale>4.999997473404071 5.00064431995692 1</scale>
              <uri>model://ocean_waves/meshes/mesh.dae</uri>
            </mesh>
          </geometry>
          <material>
            <script>
              <uri>model://ocean_waves/materials/scripts/waves.material</uri>
              <name>WaveSim/GerstnerWaves</name>
            </script>
          </material>
          <laser_retro>-1</laser_retro>
        </visual>
        <visual name="ocean_waves_below_visual">
          <pose>0 0 -0.05 0 0 0</pose>
          
          <geometry>
            <mesh>
              <scale>4.999997473404071 5.00064431995692 1</scale>
              <uri>model://ocean_waves/meshes/mesh_below.dae</uri>
            </mesh>
          </geometry>
          <material>
            <script>
              <uri>model://ocean_waves/materials/scripts/waves.material</uri>
              <name>WaveSim/GerstnerWaves</name>
            </script>
          </material>
          <laser_retro>-1</laser_retro>
        </visual>
      </link>
    </model>
    
    <plugin filename="libusv_gazebo_wind_plugin.so" name="wind">
      
      <wind_obj>
        <name>wamv</name>
        <link_name>wamv/base_link</link_name>
        <coeff_vector>.5 .5 .33</coeff_vector>
      </wind_obj>
      
      <wind_direction>270</wind_direction>
      
      <wind_mean_velocity>0</wind_mean_velocity>
      <var_wind_gain_constants>0</var_wind_gain_constants>
      <var_wind_time_constants>1</var_wind_time_constants>
      <random_seed />
      
      <update_rate>10</update_rate>
      <topic_wind_speed>/vrx/debug/wind/speed</topic_wind_speed>
      <topic_wind_direction>/vrx/debug/wind/direction</topic_wind_direction>
    </plugin>

  </world>
</sdf>